src/blockdata/transaction.rs

```diff
@@ -477,8 +544,11 @@ impl Transaction {
         }
 
         let mut engine = Sighash::engine();
-        self.encode_signing_data_to(&mut engine, input_index, script_pubkey, sighash_u32)
-            .expect("engines don't error");
+        if self.encode_signing_data_to(&mut engine, input_index, script_pubkey, sighash_u32)
+            .is_sighash_single_bug()
+            .expect("engines don't error") {
+            return Sighash::from_slice(&UINT256_ONE).expect("const-size array");
```

Comment:
`Sighash::from_inner(UINT256_ONE)` should work & avoid `expect`. I also wonder if it'd be better to make `UINT256_ONE` and associated const/method of `Sighash` intended for this purpose. Not sure how exactly to name it though.

Reply:
The name and place for `UINT256_ONE` are to be figured out in another PR. `constants` were suggested.